### ISSUE TITLE
Using Adam optimizer for online class incremental learning(CIL)

### ISSUE THREAD

**Issue by jihwankwak** [YananGu/DVC#3]
First of all, thank you for your great work and for also sharing your nicely organized code.

While I was searching several online continual learning papers, I figured out that many papers have adopted 'SGD' optimizer as a base optimizer for their implementation, which is also used in your work.

Is there any specific reason or rules in online CIL not to use other optimizers such as Adam? This question just crossed my mind after I saw an accuracy  increase of 'ER' when Adam optimizer was used in your code.

In my opinion, if there is a reason not to use Adam, I guess the idea of tracking and using the momentum of past gradients would not make sense in online scenario where data distribution is non i.i.d. (non local i.i.d also)

In short,
1) is there any reasons not to use adam in online CIL? (Even though it shows better results)
1-1) if there is, what is the reason of not using it?

Thank you again

**Comment by YananGu** [comment]
Thank you for your interest in our work. I think the main purpose of using the same optimizer is to make a more fair comparison of performance, which also allows people to focus more on the innovation of the method proposed in the paper itself. By changing the optimizer or changing the initialization method, adjusting the learning rate is more of a parameter tuning technique.
